clippy_lints/src/casts/mod.rs

```diff
@@ -419,6 +419,8 @@ impl_lint_pass!(Casts => [
 
 impl<'tcx> LateLintPass<'tcx> for Casts {
     fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>) {
+        ptr_as_ptr::check(cx, expr, &self.msrv);
```

Comment:
I think we should still check if it's from an _external_ crate?

Reply:
True, I was about to create a test case to see if it triggers from a macro in an external crate. I'll try to figure out how to do that.

Reply:
You may look into `tests/ui/auxillary/macro_rules.rs` for some examples.

Reply:
Fixed :)